callstack/org.eclipse.tracecompass.incubator.analysis.core/src/org/eclipse/tracecompass/incubator/analysis/core/reports/ReportsDataProviderFactory.java

```diff
@@ -0,0 +1,723 @@
+import org.eclipse.tracecompass.tmf.core.dataprovider.IDataProviderDescriptor;
+import org.eclipse.tracecompass.tmf.core.dataprovider.IDataProviderDescriptor.ProviderType;
+import org.eclipse.tracecompass.tmf.core.dataprovider.IDataProviderFactory;
+import org.eclipse.tracecompass.tmf.core.exceptions.TmfConfigurationException;
+import org.eclipse.tracecompass.tmf.core.model.DataProviderCapabilities;
+import org.eclipse.tracecompass.tmf.core.model.DataProviderDescriptor;
+import org.eclipse.tracecompass.tmf.core.model.tree.ITmfTreeDataModel;
+import org.eclipse.tracecompass.tmf.core.model.tree.ITmfTreeDataProvider;
+import org.eclipse.tracecompass.tmf.core.signal.TmfSignalHandler;
+import org.eclipse.tracecompass.tmf.core.signal.TmfSignalManager;
+import org.eclipse.tracecompass.tmf.core.signal.TmfTraceClosedSignal;
+import org.eclipse.tracecompass.tmf.core.signal.TmfTraceOpenedSignal;
+import org.eclipse.tracecompass.tmf.core.trace.ITmfTrace;
+import org.eclipse.tracecompass.tmf.core.trace.TmfTraceManager;
+import org.eclipse.tracecompass.incubator.internal.analysis.core.Activator;
+
+import com.google.common.collect.HashBasedTable;
+import com.google.common.collect.Table;
+
+/**
+ * Data provider factory for Reports. This factory manages different types of
+ * report providers through a registry system. It creates data provider
+ * descriptors for report configurations, validates configuration parameters,
+ * and stores configurations for each trace (or experiment).
+ *
+ * @author Kaveh Shahedi
+ */
+public class ReportsDataProviderFactory implements IDataProviderFactory, ITmfDataProviderConfigurator, IReportDataProvider {
+
+    /** Reports Provider factory ID */
+    public static final String ID = "org.eclipse.tracecompass.incubator.analysis.core.reports.reportsDataProviderFactory"; //$NON-NLS-1$
+
+    private static final String NAME = "Reports"; //$NON-NLS-1$
+    private static final String DESCRIPTION = "Configure various types of reports (text, image, etc.)"; //$NON-NLS-1$
+
+    private static final ITmfConfigurationSourceType CONFIGURATION_SOURCE_TYPE;
+    private static final String CONFIGURATION_ID = "org.eclipse.tracecompass.incubator.analysis.core.reports.config"; //$NON-NLS-1$
+    private static final String CONFIGURATION_NAME = "Reports Configurator"; //$NON-NLS-1$
+    private static final String CONFIGURATION_DESCRIPTION = "Configure custom reports"; //$NON-NLS-1$
+
+    /** Constants for configuration parameters */
+    private static final String REPORT_TYPE = "reportType"; //$NON-NLS-1$
+    private static final String REPORT_TYPE_DESC = "Should be {0}"; //$NON-NLS-1$
+    private static final String PARENT_ID = "parentId"; //$NON-NLS-1$
+    private static final String PARENT_ID_DESC = "Unique ID of the parent report. The parent must be an existing report. If not specified, the root report container will be assigned as the parent."; //$NON-NLS-1$
```

Comment:
Sorry, my previous comment about mentioning unique ID was because I understood it would be user provider (i.e. user defined ID). As we now established, the ID is generated by the back-end, then the backend will take care of the uniqueness. So, you could make the description shorter again.